<launch>
  <arg name="mav_name"    default="euroc3"/>
  <arg name="namespace"   default="$(arg mav_name)" />
  <arg name="odometry"    default="odometry" />
  <arg name="traj_in"     default="position_hold/trajectory_out" />
  <arg name="traj_out"    default="position_hold/trajectory" />

  <group ns="$(arg namespace)" >

    <node pkg="impedance_control" type="impedance_control_node" name="impedance_control" output="screen">
      <param name="rate" value="1000"/>
      <param name="moving_average_sample_number" value="20"/>
      <param name="median_filter_size" value="21"/>
      <param name="ImpedanceType" value="1"/>
      <param name="dead_zone" value="0.5"/>
      <param name="pt1_filter_time_constant" value="0.05"/>
      <param name="impedance_control_params_file" value="/config/impedance_control_params.yaml"/>
      <remap from="impedance_control/pose_command" to="command/pose"/>
      <param name="capability_group" value="ImpedanceControl"/>

      <remap from="impedance_control/odometry_meas_input"  to="$(arg odometry)"/>
      <remap from="impedance_control/trajectory_point_ref_input" to="$(arg traj_in)"/>
      <remap from="impedance_control/trajectory_point_output" to="$(arg traj_out)"/>
    </node>

  </group>

</launch>

